Redmine 2.x compatibility patch

n-rodriguez opened the pull request:
Hi there!

This is a big compatibility patch. 
With this one Redmine Git Hosting is now fully compatible with Redmine 2.x :)

MichaelKling commented:
Just tested on redmine 2.0.3 and Gitolite v3.04.

This update could fix the issues #97 and #52 (and possible some others)

Thank you very much!

n-rodriguez commented:
I confirm, this fix https://github.com/kubitron/redmine_git_hosting/issues/52

dmur commented:
:+1:

Commit:
Merge branch 'master' of github.com:jbox-web/redmine_git_hosting

Commit:
Fix https://github.com/jbox-web/redmine_git_hosting/issues/64

Commit:
Add logger class

Commit:
Use new logger class

Commit:
Call things only once

Commit:
Use new class logger

Commit:
Clean logs

Commit:
Minor change

Commit:
Fix repository hierarchy and global handling (this should fix a lot of bugs
but may introduce a major breakout in multi repo configurations)

Commit:
Clean logs

Commit:
Rename libs dir

Commit:
Remove useless hook

Commit:
Add plugin icon

Commit:
Add spaces

Commit:
Add checked_image2 method, comments, reorder method to ease merge of 0.6.2 and 0.7

Commit:
Add spaces

Commit:
Remove useless files

Commit:
Transfert reldir_add_dotslash method

Commit:
Use new logger class, validate Mirror URL, add spaces

Commit:
Remove locales

Commit:
Add images

Commit:
Add theme for JQuery Dialog boxes

Commit:
Add labels, indent, add JQuery Dialog for Rails3

Commit:
Add labels, indent, add JQuery Dialog for Rails3

Commit:
Use minimal layout for JS

Commit:
Use new logger class

Commit:
Indent

Commit:
Render JS properly

Commit:
Update Git instructions

Commit:
Add JS destroy templates

Commit:
Merge branch 'master' of github.com:jbox-web/redmine_git_hosting

Commit:
Don't be a block blocker

Commit:
Reorder things

Commit:
Add plugin link in settings

Commit:
Add :create_gitolite_ssh_key permission, reorder things

Commit:
Reindent, make ReSTful

Commit:
Merge branch 'master' of github.com:jbox-web/redmine_git_hosting

Commit:
Minor css fixes

Commit:
Load JQuery Dialog for Rails3

Commit:
Merge branch 'master' of github.com:jbox-web/redmine_git_hosting

Commit:
Rename

Commit:
Remove useless file

Commit:
Add Sidekiq worker (for the future)

Commit:
Add JQuery TagIt (for the future)

Commit:
Update locales

Commit:
Hide quoted Sidekiq routes (for the future)

Commit:
Update .gitignore

Commit:
New JS files

Commit:
Update CSS

Commit:
Use new permission

Commit:
Reorder things, add available_urls method

Commit:
Rename gl_repos method to gitolite_repos

Commit:
Rename cache_shell_redirector

Commit:
Rename class, indent

Commit:
Update partials path

Commit:
Don't be a block blocker

Commit:
Add Sidekiq libs (for the future)

Commit:
Reorder things

Commit:
Don't be a block blocker

Commit:
Define things properly

Commit:
Fix lib loading

Commit:
Rename git_adapter_hooks

Commit:
Switch to new config format (symbols instead of strings)

Commit:
Use new configuration module

Commit:
Switch to new config format and use new configuration module

Commit:
Use new configuration module

Commit:
Use new configuration module

Commit:
Use new configuration module

Commit:
Use new configuration module

Commit:
Load JQuery dialog

Commit:
Add labels, indent

Commit:
Switch to new settings views and configuration format

Commit:
Merge branch 'master' of github.com:jbox-web/redmine_git_hosting

Commit:
Reorder things, use new configuration class

Commit:
Update settings view

Commit:
Update git url partial for project overview

Commit:
Use new configuration module

Commit:
Merge branch 'master' of github.com:jbox-web/redmine_git_hosting

Commit:
Update url views

Commit:
Minor fixes

Commit:
Indent

Commit:
Reorder things

Commit:
Handle post-receive.d directory in Gitolite

Commit:
Update Rakefile

Commit:
Fix gitolite hook

Commit:
Indent

Commit:
Add comment

Commit:
Add owner and location method

Commit:
Dependencies are in Rakefile

Commit:
Minor change

Commit:
Add Sidekiq mode dependencies

Commit:
Replace repository view

Commit:
Update post receive hook

Commit:
Add Git mailing list hook

Commit:
Update labels

Commit:
Disable Sidekiq dependencies

Commit:
Rename class

Commit:
Rename class, declare method properly

Commit:
Add spaces

Commit:
Declare InstanceMethod properly

Commit:
Fix lock method args

Commit:
Finish GitAdapterHooks rename

Commit:
Remove include headers, add spaces

Commit:
Disable update for create_git_extra

Commit:
Add settings migration

Commit:
Merge branch 'master' of github.com:jbox-web/redmine_git_hosting

Commit:
Rename GitRepositoryExtra => RepositoryGitExtra

Commit:
Finish rename GitRepositoryExtra

Commit:
Add missing config method

Commit:
Rename DeploymentCredential model to RepositoryDeploymentCredential

Commit:
Rename DeploymentCredentialsViews

Commit:
Finish renaming DeploymentCredentials

Commit:
Rename form file

Commit:
Rename deployment_credentials_controller

Commit:
Finish renaming DeploymentCredentials

Commit:
Fix RepositoryDeploymentCredentials

Commit:
Add Git Mailing List

Commit:
Add Sidekiq dependencies (for the future)

Commit:
Add missing routes

Commit:
Add Sidekiq scripts (for the future)

Commit:
Indent

Commit:
Load InstanceMethods properly

Commit:
Add new columns

Commit:
Rename db migration

Commit:
Fix DeploymentCredentials renaming

Commit:
Fix DeploymentCredentials renaming

Commit:
Add missing methods (due to merge with 0.7)

Commit:
Fix do not load observers when migrating

Commit:
Fix JS modal box loading

Commit:
Fix plugin load

Commit:
Comment Sidekiq gems

Commit:
Use redmine_plugin_views_revisions to handle views

Commit:
Remove generated views

Commit:
typo

Commit:
Fix Ruby 1.8 compat

Commit:
Force remove of expired filed

Commit:
Ignore generated views

Commit:
Hide Sidekiq settings tabs

Commit:
Fix permission update on migrate

Commit:
Don't ignore error message

Commit:
Fix destroy JS rendering

Commit:
Fix permission update on migrate

Commit:
Use rev views

Commit:
Rename JQuery files

Commit:
Update .gitignore

Commit:
Fix strange bug with helper and image_tag

Commit:
Remove semicolumns

Commit:
typo

Commit:
Add rev views

Commit:
Typo

Commit:
Handle view in rev views

Commit:
Add gitolite keys view

Commit:
Add spaces

Commit:
Add spaces

Commit:
Declare InstanceMethods properly

Commit:
Update README

Commit:
Update .gitignore

Commit:
Update README

Commit:
Handle settings view in rev views

Commit:
Add default parameters for migration

Commit:
Exclude app/views/settings/_redmine_git_hosting.html.erb

Commit:
Handle app/views/settings/_gitolite_config_notify.html.erb in rev views

Commit:
JS for Git urls is framework independant

Commit:
Minor change

Commit:
JS for ZeroClipboard is framework independant

Commit:
Switch back to global git url view

Commit:
Remove Rails3 test

Commit:
Dry things

Commit:
Minor change

Commit:
Add missing translations

Commit:
Rework mirror key creation

Commit:
Add missing routes

Commit:
Fix Notify CIA

Commit:
Handle Notify CIA in JS headers view

Commit:
Add missing parameters

Commit:
Update label and urls

Commit:
Update logs

Commit:
Rename Git Http controller to Smart Http

Commit:
Minor change

Commit:
Update Smart HTTP logs

Commit:
Fix Smart HTTP

Commit:
Don't be a block blocker

Commit:
Define InstanceMethods properly

Commit:
Fix repo key generation

Commit:
Fix view name

Commit:
Rename view

Commit:
JS is framework independant

Commit:
Fix Git urls display

Commit:
Minor change

Commit:
Add missing translations

Commit:
Show Gitolite access in Redmine Config tab

Commit:
Fix public key form

Commit:
Remove Gitolite access from main view

Commit:
Update labels, add links in settings

Commit:
Move my_root_url method

Commit:
Add missing check

Commit:
Finish move of my_root_url method

Commit:
Minor change

Commit:
Minor change

Commit:
Turn off logger

Commit:
Minor change

Commit:
Fix fetch changes logger

Commit:
Remove duplicate

Commit:
Update logs

Commit:
Update README.mkd

Commit:
Don't ignore error messages

Commit:
Merge branch 'master' of github.com:jbox-web/redmine_git_hosting

Commit:
Fix alias method chain

Commit:
Fix Mirror push modal button

Commit:
Minor css fix

Commit:
Fix https://github.com/jbox-web/redmine_git_hosting/issues/72

Commit:
Fix https://github.com/jbox-web/redmine_git_hosting/issues/83

Commit:
Clean rake task

illwieckz committed:
simulate initial login with sudo -i also when testing

illwieckz committed:
a better way to test sudo, compares 'whoami' output instead of 'echo yes'

Akasch committed:
gitolite_hooks_controller: fix undefined local variable

Commit:
Merge pull request #90 from Akasch/fix_post_receive_urls

gitolite_hooks_controller: fix undefined local variable

Commit:
Fix https://github.com/jbox-web/redmine_git_hosting/issues/85

Commit:
JS popups fadeout instead of hide

Commit:
Update README.mkd

Commit:
Don't call Gitolite for nothing (this should speed up page-load when editing a project)

Commit:
Temporary disable AR callbacks

Commit:
Display select box according to permissions

Commit:
Merge branch 'master' of github.com:jbox-web/redmine_git_hosting

Commit:
Fix https://github.com/jbox-web/redmine_git_hosting/issues/91

illwieckz committed:
Merge remote branch 'upstream/master' into sudo

Commit:
Merge pull request #94 from illwieckz/sudo

sudo fix

bberawk committed:
Initial translation

bberawk committed:
Cleaned up obvious typos and removed doubled lines.

bberawk committed:
Another spellchecker pass

bberawk committed:
Did a thorough spellchecking pass

bberawk committed:
Replaced english quotes with french quotes when appropriate

bberawk committed:
Removed superfluous question marks

Commit:
Merge pull request #96 from bberawk/french_translation

French translation

bberawk committed:
Escaping javascript in url boxes

When rendering the array containing the urls and their permissions, we
escape the generated strings.

bberawk committed:
URI encoding the url in the url box

When a user copy pastes the URL in their terminal they might have
issues if the url has strange characters in it.

Commit:
Merge pull request #99 from bberawk/javascript_escape

Escaped the URLs and generated javascript in the url boxes

Commit:
Minor css fix

Commit:
Add missing JS variables

Commit:
Refactor calls to GitoliteShellWorker

Commit:
Fix french translation

Commit:
Fix Ajax forms

Commit:
Merge branch 'master' of github.com:jbox-web/redmine_git_hosting

Commit:
Refactor calls to GitoliteShellWorker

Commit:
Update copyright

Commit:
Update README

Commit:
Add LICENSE file

Commit:
Rename README

Commit:
Move sidekiq worker file

Commit:
Update README.md
